jekyll/_cci2/use-the-circleci-cli-to-split-tests.adoc

```diff
@@ -0,0 +1,189 @@
+
+[#junit-xml-reports]
+==== JUnit XML reports
+
+CircleCI requires test results to be uploaded as JUnit XML reports. The following formatting allows CircleCI to parse timing data from test results and use the data for test splitting:
+
+* The `file` attribute, either on the `<testsuite>` or `<testcase>` tag
+* The `time` attribute, on the `<testcase>` tag
+
+The following example is a snippet from an XML file with a format that CircleCI can parse:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites name="Mocha Tests" tests="3" failures="1">
+  <testsuite tests="3">
+    <testcase classname="foo1" name="ASuccessfulTest" time="10" file="src/__tests__/App.test.js" />
+    <testcase classname="foo2" name="AnotherSuccessfulTest" time="5" file="src/__tests__/App.test.js" />
+    <testcase classname="foo3" name="AFailingTest" time="1.1050" file="src/__tests__/App.test.js">
+        <failure type="NotEnoughFoo"> details about failure </failure>
+    </testcase>
+  </testsuite>
+</testsuites>
+```
+
+[#set-the-timings-type]
+==== Set the timings type
+
+The CLI attempts to autodetect whether to split by filename or classname based on the input to the `split` command. You may need to choose a different timings type depending on how your test coverage output is formatted, using the `--timings-type` option. Valid timing types are:
+
+* `filename`
+* `classname`
+* `testname`
+* `autodetect`
+
+```shell
+cat my_java_test_classnames | circleci tests split --split-by=timings --timings-type=classname
+```
+
+[#set-the-default-value-for-missing-timing-data]
+==== Set the default value for missing timing data
+
+For partially found test results, any tests with missing data is assigned a random small value. You can override this default value with the `--time-default` flag:
+
+```shell
+circleci tests glob "**/*.rb" | circleci tests split --split-by=timings --time-default=10s
+```
+
+[#download-timing-data]
+==== Download timing data
+
+If you need to manually store and retrieve timing data, use the xref:configuration-reference#storeartifacts[`store_artifacts` step].
+
+[#splitting-by-filesize]
+=== c. Split by filesize
+
+When provided with filepaths, the CLI can also split by filesize. Use the `--split-by` flag with the `filesize` split type:
+
+```shell
+circleci tests glob "**/*.go" | circleci tests split --split-by=filesize
+```
+
+[#running-split-tests]
+== 3. Run split tests
+
+Globbing and splitting tests does not actually run your tests. To combine test grouping with test execution, consider saving the grouped tests to a file, then passing this file to your test runner.
+
+```shell
+circleci tests glob "test/**/*.rb" | circleci tests split > /tmp/tests-to-run
+bundle exec rspec $(cat /tmp/tests-to-run)
+```
+
+The contents of the file `/tmp/tests-to-run` will be different in each container, based on `$CIRCLE_NODE_INDEX` and `$CIRCLE_NODE_TOTAL`.
+
+[#see-also]
+== See also
+
+* link:/parallelism-faster-jobs[Test splitting and parallelism]
```

Comment:
Is there a reason this is formatted as a link and not an xref?

Reply:
@devinmarieb hm, I think it was because I couldn't get it to work for the .md file. Let me try again